lib/drivers/tandemTslimDriver.js

```diff
@@ -1650,6 +1685,76 @@ module.exports = function (config) {
     return records;
   };
 
+  var buildAlarmRecords = function (data,records) {
+    var alarmRecords = filterLogEntries([PUMP_LOG_RECORDS.LID_ALARM_ACTIVATED,PUMP_LOG_RECORDS.LID_ALERT_ACTIVATED],data.log_records);
+    alarmRecords.forEach(function (alarmEntry) {
+      var alarmRecord = cfg.builder.makeDeviceEventAlarm()
+        .with_deviceTime(alarmEntry.deviceTime)
+        .set('index', alarmEntry.log_index);
+      cfg.tzoUtil.fillInUTCInfo(alarmRecord, alarmEntry.jsDate);
+
+      if(alarmEntry.alarm_id != null) {
+        var alarmValue = alarmEntry.alarm_id;
+        var alarmText = getAlarmName(alarmValue,ALARM_TYPES);
+
+        switch (alarmValue) {
+          case ALARM_TYPES.ALARM_OCCLUSION.value:
+            alarmRecord = alarmRecord.with_alarmType('occlusion');
+            break;
+          case ALARM_TYPES.ALARM_SECOND_OCCLUSION.value:
+            alarmRecord = alarmRecord.with_alarmType('occlusion');
+            alarmRecord = alarmRecord.with_payload({alarm_name: alarmText});
+            break;
+          case ALARM_TYPES.ALARM_AUTO_OFF.value:
+            alarmRecord = alarmRecord.with_alarmType('auto_off');
```

Comment:
So what we have to determine is whether when an alarm like this is triggered, if the suspension of insulin delivery is reflected elsewhere (e.g., in a basal rate change with `pump_suspended`) or if we have to do it ourselves. It's your happy job @gniezen to trigger these on your pump since you're not actually dependent on insulin :P 

The ones that fall into this category are: occlusions, auto_off, no_insulin, no_power, and the cartridge removed alarm definitely.